Just to verify, this is the current test on baf68d1, the commit before #16370:

```julia
$ make test-libgit2
    JULIA test/libgit2
     * libgit2              ERROR: LoadError: LoadError: MethodError: no method matching Base.LibGit2.RebaseOptions(::UInt32, ::Int32, ::Cstring, ::Base.LibGit2.MergeOptions, ::Base.LibGit2.CheckoutOptions)
Closest candidates are:
  Base.LibGit2.RebaseOptions(::Any, ::Any, ::Any, ::Any, ::Any, !Matched::Any)
  Base.LibGit2.RebaseOptions(::UInt32, ::Int32, !Matched::Int32, !Matched::Cstring, !Matched::Base.LibGit2.MergeOptions, !Matched::Base.LibGit2.CheckoutOptions)
  Base.LibGit2.RebaseOptions{T}(::Any)
 in (::Base.LibGit2.##119#121{Base.LibGit2.GitRepo,String})(::Base.LibGit2.GitReference) at ./libgit2.jl:429
 in with(::Base.LibGit2.##119#121{Base.LibGit2.GitRepo,String}, ::Base.LibGit2.GitReference) at ./libgit2/types.jl:581
 [inlined code] from ./libgit2.jl:413
 in (::##29#40)(::String) at /Users/kevin/Source/julia/test/libgit2.jl:369
 in mktempdir(::##29#40, ::String) at ./file.jl:272
 [inlined code] from ./tuple.jl:8
 in mktempdir(::Function) at ./file.jl:270
 in include_from_node1(::String) at ./loading.jl:426
 [inlined code] from ./util.jl:179
 in runtests(::String) at /Users/kevin/Source/julia/test/testdefs.jl:7
 [inlined code] from ./promotion.jl:248
 in (::Base.##279#280{##16#24,Tuple{String},Array{Any,1}})() at ./multi.jl:781
 in run_work_thunk(::Base.##279#280{##16#24,Tuple{String},Array{Any,1}}, ::Bool) at ./multi.jl:754
 in #remotecall_fetch#284(::Array{Any,1}, ::Function, ::Function, ::LocalProcess, ::String, ::Vararg{String,N}) at ./multi.jl:801
 [inlined code] from ./promotion.jl:248
 in #remotecall_fetch#286(::Array{Any,1}, ::Function, ::Function, ::Int64, ::String, ::Vararg{String,N}) at ./multi.jl:817
 in remotecall_fetch(::Function, ::Int64, ::String, ::Vararg{String,N}) at /Users/kevin/Source/julia/usr/lib/julia/sys.dylib:-1
 [inlined code] from /Users/kevin/Source/julia/test/runtests.jl:36
 in (::##15#23{Array{Any,1}})() at ./task.jl:308
while loading /Users/kevin/Source/julia/test/libgit2.jl, in expression starting on line 68
 in #remotecall_fetch#284(::Array{Any,1}, ::Function, ::Function, ::LocalProcess, ::String, ::Vararg{String,N}) at ./multi.jl:802
 [inlined code] from ./promotion.jl:248
 in #remotecall_fetch#286(::Array{Any,1}, ::Function, ::Function, ::Int64, ::String, ::Vararg{String,N}) at ./multi.jl:817
 in remotecall_fetch(::Function, ::Int64, ::String, ::Vararg{String,N}) at /Users/kevin/Source/julia/usr/lib/julia/sys.dylib:-1
 [inlined code] from /Users/kevin/Source/julia/test/runtests.jl:36
 in (::##15#23{Array{Any,1}})() at ./task.jl:308
 in sync_end() at ./task.jl:274
 [inlined code] from ./task.jl:283
 in (::##11#19)() at /Users/kevin/Source/julia/test/runtests.jl:31
 in cd(::##11#19, ::String) at ./file.jl:48
 in include_from_node1(::String) at ./loading.jl:426
 in process_options(::Base.JLOptions) at ./client.jl:263
 in _start() at ./client.jl:319
while loading /Users/kevin/Source/julia/test/runtests.jl, in expression starting on line 13
Makefile:12: recipe for target 'libgit2' failed
make[1]: *** [libgit2] Error 1
Makefile:575: recipe for target 'test-libgit2' failed
make: *** [test-libgit2] Error 2
```

--- a/base/libgit2/types.jl
+++ b/base/libgit2/types.jl
@@ -515,10 +515,11 @@ if LibGit2.version() >= v"0.24.0"
         checkout_opts::CheckoutOptions
     end
     RebaseOptions(; quiet::Cint = Cint(1),
+                    inmemory::Cint = Cint(0),
                     rewrite_notes_ref::Cstring = Cstring_NULL,
                     merge_opts::MergeOptions = MergeOptions(),
                     checkout_opts::CheckoutOptions = CheckoutOptions()) =
-        RebaseOptions(one(Cuint), quiet, Cint(0), rewrite_notes_ref, merge_opts, checkout_opts)
+        RebaseOptions(one(Cuint), quiet, inmemory, rewrite_notes_ref, merge_opts, checkout_opts)
 else
     immutable RebaseOptions
         version::Cuint

--- a/base/libgit2/types.jl
+++ b/base/libgit2/types.jl
@@ -357,7 +357,8 @@ immutable DiffDelta
 end
 DiffDelta() = DiffDelta(Cint(0), UInt32(0), UInt16(0), UInt16(0), DiffFile(), DiffFile())
 
-if LibGit2.version() >= v"0.24.0"
+# TODO: double check this when libgit2 v0.25.0 is released
+if LibGit2.version() >= v"0.25.0"
     immutable MergeOptions
         version::Cuint
         flags::Cint
@@ -365,7 +366,7 @@ if LibGit2.version() >= v"0.24.0"
         target_limit::Cuint
         metric::Ptr{Void}
         recursion_limit::Cuint
-        # default_driver::Cstring  # TODO: to be added in v"0.25.0" (?)
+        default_driver::Cstring
         file_favor::Cint
         file_flags::Cuint
     end
@@ -374,7 +375,7 @@ if LibGit2.version() >= v"0.24.0"
                    target_limit::Cuint = Cuint(200),
                    metric::Ptr{Void} = C_NULL,
                    recursion_limit::Cuint = Cuint(0),
-                   # default_driver::Cstring = Cstring_NULL,
+                   default_driver::Cstring = Cstring_NULL,
                    file_favor::Cint = Cint(Consts.MERGE_FILE_FAVOR_NORMAL),
                    file_flags::Cuint = Cuint(Consts.MERGE_FILE_DEFAULT)) =
         MergeOptions(one(Cuint),
@@ -383,7 +384,33 @@ if LibGit2.version() >= v"0.24.0"
                      target_limit,
                      metric,
                      recursion_limit,
-                     # default_driver,
+                     default_driver,
+                     file_favor,
+                     file_flags)
+elseif LibGit2.version() >= v"0.24.0"
+    immutable MergeOptions
+        version::Cuint
+        flags::Cint
+        rename_threshold::Cuint
+        target_limit::Cuint
+        metric::Ptr{Void}
+        recursion_limit::Cuint
+        file_favor::Cint
+        file_flags::Cuint
+    end
+    MergeOptions(; flags::Cint = Cint(0),
+                   rename_threshold::Cuint = Cuint(50),
+                   target_limit::Cuint = Cuint(200),
+                   metric::Ptr{Void} = C_NULL,
+                   recursion_limit::Cuint = Cuint(0),
+                   file_favor::Cint = Cint(Consts.MERGE_FILE_FAVOR_NORMAL),
+                   file_flags::Cuint = Cuint(Consts.MERGE_FILE_DEFAULT)) =
+        MergeOptions(one(Cuint),
+                     flags,
+                     rename_threshold,
+                     target_limit,
+                     metric,
+                     recursion_limit,
                      file_favor,
                      file_flags)
 else

--- a/test/libgit2.jl
+++ b/test/libgit2.jl
@@ -375,6 +375,14 @@ mktempdir() do dir
             LibGit2.branch!(repo, test_branch)
             @test_throws LibGit2.Error.GitError LibGit2.merge!(repo, fastforward=true)
 
+            # Set the username and email for the test_repo (needed for rebase)
+            cfg = LibGit2.GitConfig(repo)
+            LibGit2.set!(cfg, "user.name", "AAAA")
+            LibGit2.set!(cfg, "user.email", "BBBB@BBBB.COM")
+
+            # Try rebasing on master instead
+            LibGit2.rebase!(repo, master_branch)
+
             # Switch to the master branch
             LibGit2.branch!(repo, master_branch)
 

